pkg/test/resourcefixture/testdata/basic/gkehub/v1beta1/gkehubfeaturemembership/fullgkehubfeaturemembership/create.yaml

```diff
@@ -28,9 +29,17 @@ spec:
     configSync:
       sourceFormat: hierarchy
       git:
+        gcpServiceAccountRef:
```

Comment:
I checked the CRD and the field is named gcpServiceAccountRef([code](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_gkehubfeaturememberships.gkehub.cnrm.cloud.google.com.yaml#L79)), I think that comes from TF field([ref](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/gke_hub_feature_membership#gcp_service_account_email)).

Reply:
Right, I think this field seems to be wrong. Is there any reason (or context) we accept this field at the beginning.

Reply:
Ah my mistake, this is a DCL-based resource. 

So I think the reference field name is generated by the dcl2crd [generator](https://cnrm.git.corp.google.com/cnrm/+/refs/heads/archived/master/pkg/crd/crdgeneration/dcl2crdgeneration.go#251).  The [formatReferenceFieldName](https://cnrm.git.corp.google.com/cnrm/+/refs/heads/archived/master/pkg/dcl/extension/extension.go#155) function takes the reference field name form DCL [schema](https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/f567e1ee5f2f6293b5e204c4a0a80b861503d8d9/services/google/gkehub/feature_membership.yaml#L129), and change it to `xxxRef` format. If the field name contains name or email suffix, it will be removed([code](https://cnrm.git.corp.google.com/cnrm/+/refs/heads/archived/master/pkg/dcl/extension/extension.go#29)). That's say, `GcpServiceAccountEmail` will be converted to `GcpServiceAccountRef` by the generator.

Reply:
I think the field is a KCC IAMserviceAccount Reference